frontend/src/component/archive/ArchiveTable/ArchivedFeatureActionCell/ArchivedFeatureReviveConfirm/ArchivedFeatureReviveConfirm.tsx

```diff
@@ -0,0 +1,98 @@
+import { Alert, styled } from '@mui/material';
+import React from 'react';
+import { Dialogue } from 'component/common/Dialogue/Dialogue';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import useToast from 'hooks/useToast';
+import useProjectApi from 'hooks/api/actions/useProjectApi/useProjectApi';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+
+interface IArchivedFeatureReviveConfirmProps {
+    revivedFeatures: string[];
+    projectId: string;
+    open: boolean;
+    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+    refetch: () => void;
+}
+
+const StyledParagraph = styled('p')(({ theme }) => ({
+    marginTop: theme.spacing(2),
+}));
+
+export const ArchivedFeatureReviveConfirm = ({
+    revivedFeatures,
+    projectId,
+    open,
+    setOpen,
+    refetch,
+}: IArchivedFeatureReviveConfirmProps) => {
+    const { setToastData, setToastApiError } = useToast();
+    const { reviveFeatures } = useProjectApi();
+
+    const onReviveFeatureToggle = async () => {
+        try {
+            if (revivedFeatures.length === 0) {
+                return;
+            }
+            await reviveFeatures(projectId, revivedFeatures);
+
+            await refetch();
+            setToastData({
+                type: 'success',
+                title: "And we're back!",
+                text: 'The feature toggles have been revived.',
+            });
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        } finally {
+            clearModal();
+        }
+    };
+
+    const clearModal = () => {
+        setOpen(false);
+    };
+
+    const title = `Revive feature toggle${
```

Comment:
this should also probably be called in a test since we have logic for pluralizing here